src/docfx/build/xref/XrefResolver.cs

```diff
@@ -217,8 +221,26 @@ private static string RemoveSharingHost(string url, string hostName)
 
         private (IXrefSpec?, string? href) ResolveInternalXrefSpec(string uid, Document referencingFile, Document inclusionRoot)
```

Comment:
Consider adding an optional `MonikerList?` as a parameter, and switch behaviors based on whether the resolve carries over monikers.